components/globals/LanguageToggle.tsx

```diff
@@ -1,16 +1,23 @@
-import React from "react";
+import React, { useEffect, useRef } from "react";
 import { useTranslation } from "next-i18next";
 import { useRouter } from "next/router";
 import { StyledLink } from "./StyledLink/StyledLink";
 
 const LanguageToggle = () => {
   const { t, i18n } = useTranslation("common");
   const locale = i18n.language;
-  const { pathname } = useRouter();
+  const { pathname, asPath, isReady } = useRouter();
+  const href = useRef(pathname);
```

Comment:
If the component needs to rerender based on changes to href we may need to leverage useState() instead?